src/beast/beast/threads/SharedData.h

```diff
@@ -125,45 +126,18 @@ class SharedData
     class ConstUnlockedAccess;
 
     /** Create a shared data container.
-        Up to 8 parameters can be specified in the constructor. These parameters
-        are forwarded to the corresponding constructor in Data. If no
-        constructor in Data matches the parameter list, a compile error is
-        generated.
+        All specified parameters are forwarded to the corresponding constructor
+        in Data. If no constructor in Data matches the parameter list, a compile
+        error is generated.
     */
-    /** @{ */
-    SharedData () = default;
-
-    template <class T1>
-    explicit SharedData (T1 t1)
-        : m_value (t1) { }
-
-    template <class T1, class T2>
-    SharedData (T1 t1, T2 t2)
-        : m_value (t1, t2) { }
-
-    template <class T1, class T2, class T3>
-    SharedData (T1 t1, T2 t2, T3 t3)
-        : m_value (t1, t2, t3) { }
-
-    template <class T1, class T2, class T3, class T4>
-    SharedData (T1 t1, T2 t2, T3 t3, T4 t4)
-        : m_value (t1, t2, t3, t4) { }
-
-    template <class T1, class T2, class T3, class T4, class T5>
-    SharedData (T1 t1, T2 t2, T3 t3, T4 t4, T5 t5)
-        : m_value (t1, t2, t3, t4, t5) { }
-
-    template <class T1, class T2, class T3, class T4, class T5, class T6>
-    SharedData (T1 t1, T2 t2, T3 t3, T4 t4, T5 t5, T6 t6)
-        : m_value (t1, t2, t3, t4, t5, t6) { }
-
-    template <class T1, class T2, class T3, class T4, class T5, class T6, class T7>
-    SharedData (T1 t1, T2 t2, T3 t3, T4 t4, T5 t5, T6 t6, T7 t7) : m_value (t1, t2, t3, t4, t5, t6, t7) { }
+    template <class... Params,
+        class = std::enable_if_t<std::is_constructible<Value, Params...>{}>
```

Comment:
Not quite.  The parameter `P&&` becomes `T&&` when bound to an rvalue `T`.  In this case `P` deduced to `T`.  If instead `P&&` binds to an lvalue `T`, it becomes `T&`.  In this case `P` deduced to `T&`, and `T& &&` collapsed to `T&`.